frontend/src/components/AdminControls/Users.js

```diff
@@ -1,133 +1,172 @@
-import React, { Fragment } from 'react';
+import React, { Component, Fragment } from 'react';
 import BootstrapTable from 'react-bootstrap-table-next';
 import paginationFactory from 'react-bootstrap-table2-paginator';
 import { Type } from 'react-bootstrap-table2-editor';
 import './AdminTables.css';
+import AdminEditModal from '../AdminEditModal/AdminEditModal';
 
-const Users = ({ users }) => {
-  const columns = [
-    {
-      dataField: 'inStock', // name?
-      text: '',
-      headerStyle: {
-        borderTop: 'none',
-        borderBottom: 'none',
-      },
-      style: {
-        paddingLeft: '10px',
-      },
-      formatter: (_cellContent, row) => (
-        <div className="checkbox">
-          <label htmlFor="adminCheckbox" className="admin-checkbox-container user-checkbox-spacing">
-            <input type="checkbox" id="adminCheckbox" checked={row.inStock} />
-            <span className="admin-table-checkmark" />
-          </label>
-        </div>
-      ),
-    },
-    {
-      dataField: 'full_name',
-      text: 'User Name',
-      style: { textAlign: 'center' },
-      headerStyle: {
-        textAlign: 'center',
-        borderTop: 'none',
-        borderBottom: 'none',
-      },
-    },
-    {
-      dataField: 'created_at',
-      text: 'Member Since',
-      style: { textAlign: 'center' },
-      headerStyle: {
-        textAlign: 'center',
-        borderTop: 'none',
-        borderBottom: 'none',
-      },
-      formatter: (cell) => {
-        let dateObj = cell;
-        if (typeof cell !== 'object') {
-          dateObj = new Date(cell);
-        }
+class Users extends Component {
+  constructor(props) {
+    super(props);
 
-        const day = `0${dateObj.getUTCDate()}`.slice(-2);
-        const month = `0${dateObj.getUTCMonth() + 1}`;
-        const year = dateObj.getUTCFullYear();
-        return `${month}/${day}/${year}`;
-      },
-      editor: {
-        type: Type.DATE,
-      },
-    },
-    {
-      dataField: 'recipes',
-      text: 'Number Of Recipes',
-      align: 'center',
-      style: {
-        paddingRight: '80px',
-      },
-      headerStyle: {
-        paddingRight: '80px',
-        textAlign: 'center',
-        borderTop: 'none',
-        borderBottom: 'none',
+    this.state = {
+      editModal: false,
+      selectedUser: {},
+    };
+    this.toggleEditModal = this.toggleEditModal.bind(this);
+  }
+
+  toggleEditModal() {
+    this.setState(prevState => ({
+      editModal: !prevState.editModal,
+    }));
+  }
+
+  render() {
+    const { users } = this.props;
+    const columns = [
```

Comment:
I know you're just moving these things around for the most part, but I wonder if we could break the column definitions into a different file or something in this same directory, and then pull them in here?

Reply:
on hold based on our conversation since there are functions in the column definitions